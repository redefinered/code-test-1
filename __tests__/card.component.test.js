/* eslint-disable no-undef */

import React from 'react';
import Card from '../src/components/card/card.component';

import renderer from 'react-test-renderer';

it('renders correctly if points property is 99999', () => {
  const tree = renderer.create(<Card points={99999} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly if no points property', () => {
  const tree = renderer.create(<Card />).toJSON();
  expect(tree).toMatchSnapshot();
});
