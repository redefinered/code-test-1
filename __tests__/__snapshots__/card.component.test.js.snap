// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly if no points property 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 3,
      "flexDirection": "row",
      "height": 150,
      "marginBottom": 15,
      "marginHorizontal": 8,
      "shadowColor": "gray",
      "shadowOffset": Object {
        "height": 2,
        "width": 2,
      },
      "shadowOpacity": 0.9,
      "shadowRadius": 1,
      "width": 250,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "padding": 10,
        "width": 90,
      }
    }
  >
    <Image
      height={100}
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/item.png",
        }
      }
      width={100}
    />
  </View>
  <View
    style={
      Object {
        "justifyContent": "space-between",
        "padding": 10,
        "width": 160,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "blue",
          "fontSize": 17,
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }
    />
  </View>
</View>
`;

exports[`renders correctly if points property is 99999 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 3,
      "flexDirection": "row",
      "height": 150,
      "marginBottom": 15,
      "marginHorizontal": 8,
      "shadowColor": "gray",
      "shadowOffset": Object {
        "height": 2,
        "width": 2,
      },
      "shadowOpacity": 0.9,
      "shadowRadius": 1,
      "width": 250,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "padding": 10,
        "width": 90,
      }
    }
  >
    <Image
      height={100}
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/item.png",
        }
      }
      width={100}
    />
  </View>
  <View
    style={
      Object {
        "justifyContent": "space-between",
        "padding": 10,
        "width": 160,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "blue",
          "fontSize": 17,
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }
    />
    <View
      style={
        Object {
          "paddingBottom": 15,
        }
      }
    >
      <Text>
        Earn up to 
        <Text
          style={
            Object {
              "color": "green",
              "fontWeight": "bold",
            }
          }
        >
          99999
        </Text>
         points
      </Text>
    </View>
  </View>
</View>
`;
